### ISSUE TITLE
Record copy constructor should be protected

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#44897]
The copy constructor is currently `public` which means it can be referenced unintentionally.
```C#
record A;
record B(object P) : A;

class Program
{
    static void Main()
    {
        _ = new B(null); // NullReferenceException calling B.B(B) rather than B.B(object)
    }
}
```

**Comment by jcouv** [comment]
Relates to https://github.com/dotnet/roslyn/issues/44902 (derived copy constructor should leverage base copy constructor)

**Comment by jcouv** [comment]
Fix merged into 16.7p3. Closing this
